manifests/init.pp

```diff
@@ -205,6 +210,9 @@
   Integer[1,65536]                                     $redis_server_port,
   Integer[1]                                           $redis_timeout,
   Stdlib::Absolutepath                                 $unbound_conf_d,
+  Optional[Stdlib::Absolutepath]                       $hints_file            = "${confdir}/root.hints",
```

Comment:
There are two reasons for setting this to `Optional`:

* There could be a situation in which a user provides a root file from outside the module (i.e. a 3rd-party module). The path to that file should still be specified. Toggling `manage_roothints_file` will unmanage the file in this module, but still make it present in `unbound.conf`. 
* A user might not want to use a root hints file at all and operate solely off root hints built into Unbound. Making the `hints_file` optional and allows a user to set it `null` removing the line  from `unbound.conf`.

Reply:
My point is it has a default value as such it is not optional.

Reply:
Thanks @b4ldr, that makes sense. I was operating under the assumption one could override this hieradata with the `~` value, but this wouldn't work passing it as a class parameter. I think this could be fixed by creating a new variable `root_hints_in_config`, which I've added below.